### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by gedewon**
- [x] Make sure that there are no linter errors.
- [x] Make sure that you used correct Gitflow.
- [x] Make sure that you documented your work in a professional way.

# Project requirements #

- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. 

### Integration specs ###
**Use Capybara to write integration tests for each view in your project.**

**User index page:**

- [x] I can see the username of all other users.
- [x] I can see the profile picture for each user.
- [x] I can see the number of posts each user has written.
- [x] When I click on a user, I am redirected to that user's show page.

**user show page:**

- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can see the user's bio.
- [x] I can see the user's first 3 posts.
- [x] I can see a button that lets me view all of a user's posts.
- [x] When I click a user's post, it redirects me to that post's show page.
- [x] When I click to see all posts, it redirects me to the user's post's index page.

**User post index page:**

- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can see a post's title.
- [x] I can see some of the post's body.
- [x] I can see the first comments on a post.
- [x] I can see how many comments a post has.
- [x] I can see how many likes a post has.
- [x] I can see a section for pagination if there are more posts than fit on the view.
- [x] When I click on a post, it redirects me to that post's show page.

**Post show page:**

- [x] I can see the post's title.
- [x] I can see who wrote the post.
- [x] I can see how many comments it has.
- [x] I can see how many likes it has.
- [x] I can see the post body.
- [x] I can see the username of each commentor.
- [x] I can see the comment each commentor left.

#### Be sure to include integration specs for any other features you may have implemented! ####

### CODE REVIEW

File: spec/Integration/users/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    expect(page).to have_xpath("//img[@src='#{user.photo}']")
+  end
+  it "I can see the user's username" do
+    expect(page).to have_text(user.name)
+  end
+
+  it 'I can see the number of posts the user has written.' do
+    expect(page).to have_content(/posts:[\d+]/i)
+  end
+
```
Comment (OLIPLICHE) [inline spec/Integration/users/user_show_spec.rb:31]: - One Test is missing, Please add it:
- **Has featured tests for `User` show page:** "Tests that asserts you can see the user's first 3 posts."

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Gedewon, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/Integration/users/user_show_spec.rb (hunk 1 above)
Comment (gedewon) [reply spec/Integration/users/user_show_spec.rb:31]: @OLIPLICHE Thank you for the feedback !!

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Gedewon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**